xarray/backends/rasterio_.py

```diff
@@ -217,7 +217,11 @@ def open_rasterio(filename, chunks=None, cache=None, lock=None):
     if chunks is not None:
         from dask.base import tokenize
         # augment the token with the file modification time
-        mtime = os.path.getmtime(filename)
+        try:
+            mtime = os.path.getmtime(filename)
+        except FileNotFoundError:
```

Comment:
Python 2 only has the super-class `OSError`, so that should probably be used instead here.